### PR TITLE
Filter out empty events

### PR DESCRIPTION
**Pull request by NGoetz**
Due to sloppy implementation in #296 by @LucasConstantin and an at least equally sloppy review by myself, there were unused variables which didn't handle the case of inverted cut boundaries. Upon fixing and testing this, I noticed that we do not handle correctly empty events when filtering, namely we do not remove them. The multiplicity cut event removal was not done for constructors, only for manual filtering afterwards. This opened Pandora's box.
In the current state of affairs, I keep events which are fed into the constructor empty, and only remove such which are emptied by filtering. In future, this should maybe be customisable.

### CODE REVIEW

File: tests/test_Jetscape.py [hunk 1]
```diff
@@ -324,7 +324,7 @@ def test_filter_rapidity_in_Jetscape_constructor(tmp_path, jetscape_file_path):
     # In this test we test the integration of filters in the Jetscape constructor,
     # which are selected with keyword arguments while initializing the
     # Jetscape object.
-
+    print("Ahoi")
```
Comment (Hendrik1704) [inline tests/test_Jetscape.py:327]: Is this a necessary print?

**Review by Hendrik1704 — CHANGES_REQUESTED** [review]
Do I understand your comment correctly? If you load a file with 5 events and then filter everything out with the multiplicity_cut for example, then the number of events is 0, num_outputs_per_event is an empty list and the particle list contains only one empty list?

File: tests/test_Jetscape.py (hunk 1 above)
Comment (NGoetz) [reply tests/test_Jetscape.py]: Removed

**Comment by NGoetz** [comment]
> Do I understand your comment correctly? If you load a file with 5 events and then filter everything out with the multiplicity_cut for example, then the number of events is 0, num_outputs_per_event is an empty list and the particle list contains only one empty list?

Yes

**Review by Hendrik1704 — APPROVED** [review]

**Pull request merged**